worker/worker.py

```diff
@@ -61,7 +62,12 @@ def run(self):
         while self._should_run():
             try:
                 self._checkin()
+                if self._last_checkin_successful == False:
```

Comment:
nit: pep8 says that this should probably be `not self._last_checkin_successful`

(Though they recommend doing the same for non-booleans, which I don't like.)